src/App.tsx

```diff
@@ -1,61 +1,150 @@
+import { TodoList } from './components/TodoList';
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+}
+
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const users = [...usersFromServer];
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState<null | number>(null);
+  const [myTodos, setMyTodos] = useState(todos);
+  const [isWrongUser, setIsWrongUser] = useState(false);
+  const [isEmptyTitle, setIsEmptyTitle] = useState(false);
+
+  const addTodo = () => {
+    const arr = myTodos.sort(
+      (todo1, todo2) => todo1.id - todo2.id,
+    );
+
+    const largestId = arr[arr.length - 1].id;
+
+    if (userId) {
+      setMyTodos(state => (
+        [...state, {
+          id: largestId + 1,
+          title,
+          completed: false,
+          userId,
+          user: getUser(userId),
+        }]
+      ));
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (title !== '' && userId === null) {
+      setIsWrongUser(true);
+    } else if (!isEmptyTitle && userId !== null && title.trim() !== '') {
+      addTodo();
+      setTitle('');
+      setUserId(null);
+    } else if (title === '' || userId === null) {
+      setIsEmptyTitle(true);
+      setIsWrongUser(true);
+    } else if (title.trim() === '') {
+      setIsEmptyTitle(true);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+          <input
+            type="text"
+            id="title"
+            data-cy="titleInput"
+            placeholder="Please enter a title"
+            value={isEmptyTitle ? '' : title}
+            onChange={(event) => {
+              setTitle(event.target.value);
+              setIsEmptyTitle(false);
+            }}
+          />
+
+          {isEmptyTitle
+            && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <label htmlFor="select">User: </label>
+          <select
+            data-cy="userSelect"
+            id="select"
+            value={userId !== null ? userId : 0}
```

Comment:
```suggestion
            value={userId ?? 0}
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment